docs/patterns.md

```diff
@@ -0,0 +1,138 @@
+# Patterns in Tungsten.js Applications
+
+The following are some general patterns and principles for building Tungsten.js applications.  As with all patterns, there are exceptions and they are likely to change over time.  But these are generally good starting points when building an application.
+
+## Events up, methods down
+
+The most important principle for dealing with communciation between models in Tungsten.js is this: communication up the model tree should be done via events, communication down the model tree should be done by calling methods on the models.  Events up, methods down.
+
+What does this mean?  Suppose there are two models in a collection that need to communicate with one another.  Instead of the communication occuring directly between these two sibling models, the communication is going to go through their nearest ancestor, which in this case would be the parent model of the collection.  To do this, one model will trigger an event, e.g., `this.trigger('someEvent', data);` and the parent model will listen to that event and call a method on the second model to communicate the necessary information.  It will find which model in the collection to call the method on likely via a `collection.findWhere(attrs)` call.  Ideally the event and method should pass through only the information necessary for the communication and no more.  This is a simple case of events up, methods down for communicating between models, but it scales to more complex cases as well.
```

Comment:
Or this could go down in the 'Trigger and listen to model, not view, events' section